src/sync/sync_permission.cpp

```diff
@@ -213,28 +278,19 @@ void Permissions::set_permission(std::shared_ptr<SyncUser> user,
         if (auto code = any_cast<long long>(status_code)) {
             // The permission change failed because an error was returned from the server.
             auto status = object->get_property_value<util::Any>(context, "statusMessage");
-            std::string error_str = status.has_value()
-                                  ? any_cast<std::string>(status)
-                                  : util::format("Error code: %1", code);
-            callback(std::make_exception_ptr(PermissionChangeException(error_str, code)));
+            std::string error_str = (status.has_value()
+                                     ? any_cast<std::string>(status)
+                                     : util::format("Error code: %1", code));
+            handler(nullptr, std::make_exception_ptr(PermissionActionException(error_str, code)));
         }
         else {
-            callback({});
+            handler(&*object, nullptr);
```

Comment:
Isn't `&*` here equivalent to `object.get()`?

Reply:
Ah, that's it.